x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -211,11 +245,165 @@ export const AlertsTableComponent: React.FC<AlertsTableComponentProps> = ({
+
+  const queryGroups = useMemo(
+    () =>
+      getAlertsGroupingQuery({
+        additionalFilters,
+        selectedGroup,
+        from,
+        runtimeMappings,
+        to,
+        pageSize: pagination.pageSize,
+        pageIndex: pagination.pageIndex,
+      }),
+    [
+      additionalFilters,
+      selectedGroup,
+      from,
+      runtimeMappings,
+      to,
+      pagination.pageSize,
+      pagination.pageIndex,
+    ]
+  );
+
+  const {
+    data: alertsGroupsData,
+    loading: isLoadingGroups,
+    refetch,
+    request,
+    response,
+    setQuery: setAlertsQuery,
+  } = useQueryAlerts<
+    {},
+    GroupingTableAggregation &
+      Record<string, { value?: number | null; buckets?: Array<{ doc_count?: number | null }> }>
+  >({
+    query: queryGroups,
+    indexName: signalIndexName,
+    queryName: ALERTS_QUERY_NAMES.ALERTS_GROUPING,
+    skip: !selectedGroup || selectedGroup === 'none',
+  });
+
+  useEffect(() => {
+    if (selectedGroup) {
+      setAlertsQuery(queryGroups);
+    }
+  }, [queryGroups, selectedGroup, setAlertsQuery]);
+
+  useInspectButton({
+    deleteQuery,
+    loading: isLoadingGroups,
+    response,
+    setQuery,
+    refetch,
+    request,
+    uniqueQueryId,
+  });
+
+  const inspect = useMemo(
+    () => <InspectButton queryId={uniqueQueryId} inspectIndex={0} title={''} />,
+    [uniqueQueryId]
+  );
+
+  const defaultGroupingOptions = getDefaultGroupingOptions(tableId);
+  const [options, setOptions] = useState(
+    defaultGroupingOptions.find((o) => o.key === selectedGroup)
+      ? defaultGroupingOptions
+      : [
+          ...defaultGroupingOptions,
+          ...(selectedGroup
+            ? [
+                {
+                  key: selectedGroup,
+                  label: selectedGroup,
+                },
+              ]
+            : []),
+        ]
+  );
+
+  const groupsSelector = useMemo(
+    () => (
+      <GroupsSelector
+        groupSelected={selectedGroup}
+        data-test-subj="alerts-table-group-selector"
+        onGroupChange={(groupSelection?: string) => {
+          storage.set(`${ALERTS_TABLE_GROUPS_SELECTION_KEY}-${tableId}`, groupSelection);
+          setGroupsActivePage(0);
+          setSelectedGroup(groupSelection);
+          if (
+            groupSelection &&
+            groupSelection !== 'none' &&
+            !options.find((o) => o.key === groupSelection)
+          ) {
+            setOptions([
+              ...defaultGroupingOptions,
+              {
+                label: groupSelection,
+                key: groupSelection,
+              },
+            ]);
+          } else {
+            setOptions(defaultGroupingOptions);
+          }
+        }}
+        onClearSelected={() => {
+          setGroupsActivePage(0);
+          setSelectedGroup(undefined);
+          setOptions(defaultGroupingOptions);
+          storage.set(`${ALERTS_TABLE_GROUPS_SELECTION_KEY}-${tableId}`, undefined);
+        }}
+        fields={indexPatterns.fields}
```

Comment:
Just a heads up, as we've discussed with https://github.com/elastic/kibana/issues/149533, this will have a similar initialization issue with regards to the alerts data-view:

<img width="649" alt="Alerts_-_Kibana" src="https://user-images.githubusercontent.com/2946766/216427076-002c8acb-1498-4786-93bd-7e842defd235.png">

Low impact since only present on initial setup, and could also be resolved with the reintroduction of the 'Refresh toast'.